docs/plugin_auth_keycloak.md

```diff
@@ -0,0 +1,6 @@
+**Proposed Oath2-inspired Protocol**
```

Comment:
This is a great, first PR! Thank you for contributing!

Reply:
A few, very minor suggestions. Let's use markdown format: https://www.markdownguide.org/cheat-sheet/ 
The document should start with Heading, therefore please change the first line as follow:

```
# Proposed OAth2-inspired Protocol

Currently, we are ... 
```